src/common/commonutils/DeviceInfoUtils.c

```diff
@@ -947,4 +947,22 @@ bool IsCommodore(void* log)
     FREE_MEMORY(textResult);
 
     return status;
-}
\ No newline at end of file
+}
+
+enum SelinuxState {
+  SelinuxUnknown = 0,
+  SelinuxFound,
+  SelinuxNotFound,
+};
+static enum SelinuxState g_selinuxState = SelinuxUnknown;
+
+bool DetectSelinux(void* log)

```

Comment:
>DetectSelinux [](http://example.com/codeflow?start=5&length=13)

Together with making the enum a Boolean, we could also rename the function as 'IsSelinuxPresent' to better describe its purpose. Thoughts?

Reply:
That's ok, but we need to avoid calling `CheckTextIsFoundInFile` on every invocation (it spams the log). With only one function we'd need to rely on a function static variable because this alone wouldn't work:
```
bool IsSelinuxPresent(void* log)
{
    if (!g_selinuxPresent)
    {
        g_selinuxPresent = (0 == CheckTextIsFoundInFile("/sys/kernel/security/lsm", "selinux", NULL, log));
    }
    return g_selinuxPresent;
}
```

 So I split the code into a function that does probing `DetectSelinux` and one that returns the status `IsSelinuxPresent`. Let me know if you see a different way?